src/main/scala/fi/aalto/cs/apluscourses/intellij/Repl.scala

```diff
@@ -31,6 +32,15 @@ class Repl(module: Module) extends ScalaLanguageConsole(module: Module) {
     }
   }, AWTEvent.FOCUS_EVENT_MASK)
 
+  // We need this here because the overridden ConsoleExecuteAction needs to determine whether
+  // the console is hosting a Scala 3 REPL or something else
+  val isScala3REPL: Boolean = ModuleUtils.nonEmpty(
+    ModuleRootManager.getInstance(module)
+      .orderEntries()
+      .librariesOnly()
+      .satisfying(x => x.getPresentableName.contains("scala3-") || x.getPresentableName.contains("scala-sdk-3."))
```

Comment:
This seems a bit error-prone, although I see that it may be the only way to resolve the Scala version. I wonder if it would be a good idea to better isolate this part due to its problematic nature?

Reply:
It doesn't really *need* to work in all potential cases for all projects; it just needs to work for projects created by the plugin. I feel like this is good enough for this purpose, but I wouldn't extract this code anywhere else precisely because I don't want to convey the message that the method works for all modules/projects and is therefore ready to use in other places in code.